packages/vaadin-upload/src/vaadin-upload.js

```diff
@@ -512,6 +514,13 @@ class UploadElement extends ElementMixin(ThemableMixin(PolymerElement)) {
     return maxFiles >= 0 && numFiles >= maxFiles;
   }
 
+  /** @private */
+  _maxFilesReachedChanged(maxFilesReached) {
+    maxFilesReached
+      ? this.setAttribute('max-files-reached', maxFilesReached)
+      : this.removeAttribute('max-files-reached');
```

Comment:
Should we use `reflectToAttribute` instead?

Reply:
Great suggestion, fixed.